odoo/addons/base/models/res_company.py

```diff
@@ -115,6 +116,11 @@ def _get_company_address_update(self, partner):
         return dict((fname, partner[fname])
                     for fname in self._get_company_address_field_names())
 
+    @api.depends('country_id')
+    def _compute_state_required(self):
+        for company in self:
+            company.is_state_required = company.country_id.is_state_required
+
```

Comment:
You could use a related field instead, no?

Reply:
@kba-odoo , 
country_id is non-stored, compute field with inverse method in company based on partner_id so we can not create a related field directly. 
But if we wants to make the related field of country_id then we have to add the search method on country_id field. so we preferred to make directly compute instead of define the search on country_id.
is it ok to you OR should we add the search method on country_id?

Thanks

Reply:
@mba-odoo no that's fine like that then :-) Didn't notice that country_id was not stored, my bad!